jems-testing/src/main/java/org/dmfs/jems/hamcrest/matchers/StackTopMatcher.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.hamcrest.matchers;
+
+import org.dmfs.jems.stack.Stack;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+
+/**
+ * A {@link Matcher} which matches {@link Stack.StackTop}s.
+ *
+ * @author Marten Gajda
+ */
+public final class StackTopMatcher<E> extends TypeSafeDiagnosingMatcher<Stack.StackTop<E>>
```

Comment:
This is not complex, but I think this could also be simpler with `FeatureMatcher`, and here it would actually fit properly, because `StackTop` has 2 'properties/features': `element()` and `bottom()`.

Like:
```java
class StackTopMatcherElementMatcher<T> extends FeatureMatcher<Stack.StackTop<T>, T>
{

    public StackTopMatcherElementMatcher(Matcher<? super T> elementMatcher)
    {
        super(elementMatcher, "has element()", "had element()");
    }


    @Override
    protected T featureValueOf(Stack.StackTop<T> actual)
    {
        return actual.element();
    }
}
```
And another one for `bottom()` and then combining them with `any(..)`.

Not important of course.

Reply:
Right that would be more in line with SRP.